geomstats/geometry/poincare_ball.py

```diff
@@ -317,3 +317,268 @@ def inner_product_matrix(self, base_point=None):
         identity = gs.eye(self.dimension, self.dimension)
 
         return gs.einsum('i,jk->ijk', lambda_base, identity)
+
+
+NORMALIZATION_FACTOR_CST = gs.sqrt(gs.pi / 2)
+PI_2_3 = pow((2 * gs.pi), 2 / 3)
+SQRT_2 = gs.sqrt(2.)
+
+
+class GaussianDistribution():
+    """A class for Gaussian distributions."""
+
+    @staticmethod
+    def gaussian_pdf(data, means, variances, norm_func, metric):
+        """Return the probability density function."""
+        data_length, _, n_gaussian = data.shape + (means.shape[0],)
+
+        data_expanded = gs.expand_dims(data, 1)
+        data_expanded = gs.repeat(data_expanded, n_gaussian, axis=1)
+
+        means_expanded = gs.expand_dims(means, 0)
+        means_expanded = gs.repeat(means_expanded, data_length, axis=0)
+
+        variances_expanded = gs.expand_dims(variances, 0)
+        variances_expanded = gs.repeat(variances_expanded, data_length, 0)
+
+        num = gs.exp(-(metric.dist(data_expanded, means_expanded) ** 2)
+                     / (2 * variances_expanded ** 2))
+
+        den = norm_func(variances)
+
+        den = gs.expand_dims(den, 0)
+        den = gs.repeat(den, data_length, axis=0)
+
+        result = num / den
+
+        return result
+
+    @staticmethod
+    def weighted_gmm_pdf(mixture_coefficients,
+                         mesh_data,
+                         means,
+                         variances,
+                         metric):
+        """Return the probability density function of a GMM."""
+        mesh_data_units = gs.expand_dims(mesh_data, 1)
+
+        mesh_data_units = gs.repeat(mesh_data_units, len(means), axis=1)
+
+        means_units = gs.expand_dims(means, 0)
+
+        means_units = gs.repeat(means_units, mesh_data_units.shape[0], axis=0)
+
+        distance_to_mean = metric(mesh_data_units, means_units)
+        variances_units = gs.expand_dims(variances, 0)
+        variances_units = gs.repeat(variances_units,
+                                    distance_to_mean.shape[0], axis=0)
+
+        distribution_normal = gs.exp(-(distance_to_mean ** 2)
+                                     / (2 * variances_units ** 2))
+
+        zeta_sigma = PI_2_3 * variances
+        zeta_sigma = zeta_sigma * gs.exp((variances ** 2 / 2)
+                                         * gs.erf(variances / gs.sqrt(2)))
+
+        result_num = gs.expand_dims(mixture_coefficients, 0)
+        result_num = gs.repeat(result_num,
+                               len(distribution_normal), axis=0)
+        result_num = result_num * distribution_normal
+        result_denum = gs.expand_dims(zeta_sigma, 0)
+        result_denum = gs.repeat(result_denum,
+                                 len(distribution_normal), axis=0)
+
+        result = result_num / result_denum
+
+        return result
+
```

Comment:
Since the GaussianDistribution class has only two methods, and they are static methods, I wonder if we need a class at this point? These work for the poincare ball only, right? Maybe we can have them as function there, they would be at:
`geomstats.geometry.poincare_ball.gaussian_pdf` which is fine, no need for another layer of class?

Reply:
Or we could have them as methods of the PoincareBall, since we already have random_uniform methods in the manifold classes anyway...
... and maybe later create a folder `probability` along with `geometry` and `learning`?